tests/src/traffic.rs

```diff
@@ -627,45 +628,108 @@ mod traffic_tests {
         assert!(res.success());
     }
 
+    /// Verifies that the user application can make requests to some publicly available servers.
+    ///
+    /// Runs with both enabled and disabled outgoing feature.
+    ///
+    /// # Note on flakiness
+    ///
+    /// Because the outcome of this test depends on availability of the servers we
+    /// can't control, we first check in the test code whether they're actually available.
+    ///
+    /// We take the first 10 live servers and pass them to the app.
+    /// If we don't have at least 10, we fail.
+    #[cfg_attr(not(any(feature = "ephemeral", feature = "job")), ignore)]
     #[rstest]
+    #[case::outgoing_enabled(true)]
+    #[case::outgoing_disabled(false)]
     #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
-    #[cfg_attr(not(any(feature = "ephemeral", feature = "job")), ignore)]
-    pub async fn test_outgoing_traffic_many_requests_enabled(#[future] service: KubeService) {
-        let service = service.await;
-        let node_command = vec![
-            "node",
-            "node-e2e/outgoing/test_outgoing_traffic_many_requests.mjs",
+    pub async fn test_outgoing_traffic_many_requests(
+        #[future] service: KubeService,
+        #[case] outgoing_enabled: bool,
+    ) {
+        const HOSTS: &[&str] = &[
+            "www.rust-lang.org",
+            "www.github.com",
+            "www.google.com",
+            "www.bing.com",
+            "www.yahoo.com",
+            "www.twitter.com",
+            "www.microsoft.com",
+            "www.youtube.com",
+            "www.live.com",
+            "www.msn.com",
+            "www.google.com.br",
+            "www.yahoo.co.j",
+            "www.amazon.com",
+            "www.wikipedia.org",
+            "www.facebook.com",
+            "www.drive.google.com",
+            "www.gmail.com",
+            "www.twitch.tv",
+            "www.oracle.com",
+            "www.uber.com",
         ];
-        let mut process = run_exec_with_target(
-            node_command,
-            &service.pod_container_target(),
-            None,
-            None,
-            None,
-        )
-        .await;
 
-        let res = process.child.wait().await.unwrap();
-        assert!(res.success());
-        process.assert_no_error_in_stderr().await;
-    }
+        let client = reqwest::Client::builder()
+            .connect_timeout(Duration::from_secs(5))
+            .read_timeout(Duration::from_secs(5))
+            .default_headers(HeaderMap::from_iter([(
+                USER_AGENT,
+                HeaderValue::from_static(concat!(
+                    env!("CARGO_PKG_NAME"),
+                    "/",
+                    env!("CARGO_PKG_VERSION"),
+                )),
+            )]))
```

Comment:
```suggestion
            .user_agent(concat!(
                  env!("CARGO_PKG_NAME"),
                  "/",
                  env!("CARGO_PKG_VERSION"),
             ))
```

Request has a shorthand for it